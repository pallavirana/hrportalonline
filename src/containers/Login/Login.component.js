
import React, { useState } from "react";
import withStyles from 'react-jss';
import { Formik } from "formik";
import * as yup from 'yup'
import styles from './Login.styles';
import {FormInput, FormSubmitButton, SuccessMsgBlock} from '../../components/FormElements';

const userSchema = yup.object().shape({
    email: yup
      .string()
      .email("Please Enter an valid Email")
      .required("Email is Required."),
    password: yup
      .string()
      .required("Password is Required.")
      .max(13,"Too long")
      .min(8,"Too short")
  })

const initialState = {
    email: "",
    password: ""
  };

export const Login = (props) => { 
    const { classes} = props;
    const [user, setUser] = useState(initialState);
    return (
    <div className={classes.loginWrap}>
        {user.email && <SuccessMsgBlock msg={`${user.email} loggedin succesfully`} />}
        <Formik
            initialValues={user}
            onSubmit={(values, actions) => {
            console.log(values);
            actions.setSubmitting(true);
            setUser(values);
            setTimeout(() => {
                actions.resetForm(initialState);
                actions.setSubmitting(false);
            }, 2000);
            }}
            validationSchema={userSchema}
        >
            {props => (
                <form onSubmit={props.handleSubmit} className={classes.loginForm}>
                    <FormInput
                        name="email"
                        onChange={props.handleChange}
                        value={props.values.email}
                        type="text"
                        placeholder="Email"
                        errorStatus = {props.errors.email && props.touched.email}
                        errorMsg = {props.errors.email}
                    />
                    <FormInput
                        type="password"
                        onChange={props.handleChange}
                        name="password"
                        value={props.values.password}
                        placeholder="Password"
                        errorStatus = {props.errors.password && props.touched.password}
                        errorMsg = {props.errors.password}
                    />
                    <FormSubmitButton
                        value="Login"
                        type="submit"
                        disabled={!props.dirty && !props.isSubmitting}
                    />
                </form>
            )}
        </Formik>
    </div>
)};

export default withStyles(styles)(Login);