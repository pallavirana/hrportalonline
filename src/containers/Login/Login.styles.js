export default {
    loginWrap: {
      width: '300px',
      display: 'flex',
      flexWrap: 'wrap',
      alignItems: 'flex-start',
      flexDirection: 'column',
      margin: '50px auto',
    },
    loginForm: {
        width: '100%',
    }
  };
  