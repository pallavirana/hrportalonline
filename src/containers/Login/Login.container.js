import React from 'react';
import withStyles from 'react-jss';
import styles from './Login.styles';
import { FirebaseContext } from '../../utils/Firebase';
import {LoginForm} from './Login.component';

const Login = (props) => (
  <div>
    <FirebaseContext.Consumer>
      {firebase => <LoginForm firebase={firebase} {...props} />}
    </FirebaseContext.Consumer>
  </div>
);

export default withStyles(styles)(Login);
