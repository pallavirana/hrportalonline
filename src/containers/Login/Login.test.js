import React from 'react';
import { shallow } from 'enzyme';
import {LoginForm} from './Login.component';


describe('Login', () => {
    const wrapper = shallow(<LoginForm classes={{}} />);
    it('shallow renders without crashing', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
