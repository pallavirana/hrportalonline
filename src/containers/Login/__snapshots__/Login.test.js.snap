// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login shallow renders without crashing 1`] = `ShallowWrapper {}`;
